src/ticketing/firestore.ts

```diff
@@ -20,16 +20,76 @@ export function setupFirestoreListeners(
     customer: SnapshotListener<CustomerProfile>;
   },
 ) {
+  const mapTravelRight = (travelRight: FirebaseFirestoreTypes.DocumentData): FirebaseFirestoreTypes.DocumentData => {
+    if (travelRight.startDateTime && travelRight.endDateTime) {
+      const trStartDateTime = travelRight.startDateTime as FirebaseFirestoreTypes.Timestamp;
+      const trEndDateTime = travelRight.endDateTime as FirebaseFirestoreTypes.Timestamp;
+
+      travelRight.startDateTime = trStartDateTime.toDate();
+      travelRight.endDateTime = trEndDateTime.toDate();
+    }
+
+    if (travelRight.usedAccesses) {
+      travelRight.usedAccesses = travelRight.usedAccesses.map(mapUsedAccesses);
+    }
+
+    return travelRight;
```

Comment:
This looks pretty good, just make sure that the mapping functions doesn't modify the input parameter. Instead return a new object.

https://dev.to/smeijer/stop-mutating-in-map-reduce-and-foreach-58bf